packages/worker/src/core.ts

```diff
@@ -81,20 +93,59 @@ export class Core {
     }
   }
 
-  protected isForwardingLogs(): boolean {
-    return this.options.telemetryOptions.logForwardingLevel !== 'OFF';
+  /**
+   * Instantiate a new Core object and set it as the singleton instance
+   *
+   * If Core has already been instantiated with {@link instance} or this method,
+   * will throw a {@link IllegalStateError}.
+   */
+  public static async install(options: CoreOptions): Promise<Core> {
+    // Remember the provided options in case Core is reinstantiated after being shut down
+    this.defaultOptions = options;
+    if (this._instance !== undefined) {
+      if (this.instantiator === 'install') {
+        throw new IllegalStateError('Core singleton has already been installed');
+      } else if (this.instantiator === 'instance') {
+        throw new IllegalStateError(
+          'Core singleton has already been instantiated. Did you start a Worker before calling `install`?'
+        );
+      }
+    }
+    this.instantiator = 'install';
+    this._instance = this.create(options).catch((err) => {
+      // Unset the singleton in case creation failed
+      delete this._instance;
+      throw err;
+    });
+    return this._instance;
   }
 
   /**
-   * Default options get overridden when Core is installed and are remembered in case Core is
-   * re-instantiated after being shut down
+   * Get or instantiate the singleton Core object
+   *
+   * If Core has not been instantiated with {@link install} or this method,
+   * a new Core instance will be installed and configured to connect to
+   * the local docker compose setup.
```

Comment:
I'd use the chance to rephrase this to:
```suggestion
   * a local server.
```